### PR TITLE
fix: Avatar falling back to children on image error

### PR DESCRIPTION
**Pull request by sosohime**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Perfermance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
https://github.com/ant-design/ant-design/issues/24897

### 💡 Background and solution
当image加载失败时，如果有children，则fall back到children

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Avatar falling back to children on image error      |
| 🇨🇳 Chinese |    当image加载失败时，如果有children，则fall back到children       |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/avatar/index.tsx [hunk 1]
```diff
@@ -99,7 +99,11 @@ export default class Avatar extends React.Component<AvatarProps, AvatarState> {
     const { onError } = this.props;
     const errorFlag = onError ? onError() : undefined;
     if (errorFlag !== false) {
-      this.setState({ isImgExist: false });
+      this.setState({ isImgExist: false }, () => {
+        if (this.props.children) {
```
Comment (afc163) [inline components/avatar/index.tsx:103]: children 统一从上面 this.props 引出来。

**Comment by afc163** [comment]
来个用例试试。

**Comment by afc163** [comment]
冲突了

File: components/avatar/__tests__/Avatar.test.js [hunk 2]
```diff
@@ -99,6 +99,7 @@ describe('Avatar Render', () => {
 
     expect(wrapper.find(Avatar).instance().state.isImgExist).toBe(false);
     expect(wrapper.find('.ant-avatar-string').length).toBe(1);
+    expect(wrapper.find('.ant-avatar-string').prop('style')).not.toHaveProperty('opacity', 0);
```
Comment (xrkffgg) [inline components/avatar/__tests__/Avatar.test.js:102]: 这个判断啥？

Comment (sosohime) [reply components/avatar/__tests__/Avatar.test.js:103]: 在这个用例下，当图片加载失败时，children应该被展示出来

**Comment by sosohime** [comment]
重提一个

**Comment by xrkffgg** [comment]
rebase 后 push 加 -f

**Pull request closed without merging**